storage-proofs/src/stacked/proof_scheme.rs

```diff
@@ -0,0 +1,128 @@
+use rayon::prelude::*;
+
+use crate::drgraph::Graph;
+use crate::error::Result;
+use crate::hasher::Hasher;
+use crate::proof::ProofScheme;
+use crate::stacked::{
+    challenges::ChallengeRequirements,
+    graph::StackedBucketGraph,
+    params::{PrivateInputs, Proof, PublicInputs, PublicParams, SetupParams},
+    proof::StackedDrg,
+};
+
+impl<'a, 'c, H: 'static + Hasher> ProofScheme<'a> for StackedDrg<'c, H> {
+    type PublicParams = PublicParams<H>;
+    type SetupParams = SetupParams;
+    type PublicInputs = PublicInputs<<H as Hasher>::Domain>;
+    type PrivateInputs = PrivateInputs<H>;
+    type Proof = Vec<Proof<H>>;
+    type Requirements = ChallengeRequirements;
+
+    fn setup(sp: &Self::SetupParams) -> Result<Self::PublicParams> {
+        let graph = StackedBucketGraph::<H>::new_stacked(
+            sp.drg.nodes,
+            sp.drg.degree,
+            sp.drg.expansion_degree,
+            0,
+            sp.drg.seed,
+        );
+
+        Ok(PublicParams::new(graph, sp.layer_challenges.clone()))
+    }
+
+    fn prove<'b>(
+        pub_params: &'b Self::PublicParams,
+        pub_inputs: &'b Self::PublicInputs,
+        priv_inputs: &'b Self::PrivateInputs,
+    ) -> Result<Self::Proof> {
+        let proofs = Self::prove_all_partitions(pub_params, pub_inputs, priv_inputs, 1)?;
```

Comment:
Why always one partition? I'm probably not following the logic here — which may or may not be a sign that this is either wrong or too tricky for its own good.

Reply:
Okay, I am fresh now. I will look more closely.

Reply:
I figured this out. It was a legacy hack to make `prove` match the trait and should only have been used with a single partition. I think it may have been jammed in on short notice for API reasons at one point. I'll submit a small PR making the issue more explicit, and you can propagate to this one.